gazebo/Server.cc

```diff
@@ -402,9 +405,36 @@ bool Server::LoadFile(const std::string &_filename,
     return false;
   }
 
-  if (!sdf::readFile(common::find_file(_filename), sdf))
+  std::string filename = _filename;
+  auto filenameUri = ignition::common::URI(filename);
+
+  if (filenameUri.Scheme() == "http" || filenameUri.Scheme() == "https")
   {
-    gzerr << "Unable to read sdf file[" << _filename << "]\n";
+    std::string downloadedDir = common::FuelModelDatabase::Instance()->ModelPath(filename);
+    // Find the first sdf file in the world path for now, the later intention is
+    // to load an optional world config file first and if that does not exist,
+    // continue to load the first sdf file found as done below
+    for (ignition::common::DirIter file(downloadedDir);
+         file != ignition::common::DirIter(); ++file)
+    {
+      std::string current(*file);
+      if (ignition::common::isFile(current))
+      {
+        std::string fileName = ignition::common::basename(current);
+        std::string::size_type fileExtensionIndex = fileName.rfind(".");
+        std::string fileExtension = fileName.substr(fileExtensionIndex + 1);
+
+        if (fileExtension == "sdf" || fileExtension == "world")
+        {
+          filename = current.c_str();
+        }
+      }
+    }
+  }
+
+  if (!sdf::readFile(common::find_file(filename), sdf))
```

Comment:
You've modified `common::find_file` to look into Fuel. So it's a bit strange to also look into Fuel before calling `common::find_file`. Do we need the extra logic outside this function? What if someone tries to find a Fuel world file with `find_file`? I think they should be able to get it.

Reply:
updated logic https://github.com/osrf/gazebo/pull/2822/commits/9bad120e6260130351f7f5f86fe334db84b4dad5